File: native/V2StyxLib/lib/messages/base/StyxRSingleQIDMessage.cpp
/*
 * StyxRSingleQIDMessage.cpp
 *
 *  Created on: Apr 07, 2018
 *      Author: vschryabets@gmail.com
 */

#include "messages/base/StyxRSingleQIDMessage.h"
#include <sstream>

StyxRSingleQIDMessage::StyxRSingleQIDMessage(MessageTypeEnum type, StyxTAG tag, const StyxQID* qid) :
	StyxMessage(type, tag) {
	if (qid != NULL) {
		mQID = new StyxQID(qid->getType(), qid->getVersion(), qid->getPath());
	}
}

StyxRSingleQIDMessage::~StyxRSingleQIDMessage() {
	if (mQID != NULL) {
		delete mQID;
		mQID = NULL;
	}
}

void StyxRSingleQIDMessage::load(IStyxDataReader *buffer) {
	StyxMessage::load(buffer);
	mQID = new StyxQID(buffer);
}

void StyxRSingleQIDMessage::writeToBuffer(IStyxDataWriter *outputBuffer) {
	StyxMessage::writeToBuffer(outputBuffer);
	mQID->writeBinaryTo(outputBuffer);
}

size_t StyxRSingleQIDMessage::getBinarySize() {
	return StyxMessage::getBinarySize() + StyxQID::CONTENT_SIZE;
}

StyxQID* StyxRSingleQIDMessage::getQID() {
	return mQID;
}

StyxString StyxRSingleQIDMessage::toString() {
	std::stringstream stream;
	stream << StyxMessage::toString() << "\QID: "
			<< mQID->toString();
	return stream.str();

}

File: java/v2styx-lib/src/main/java/com/v2soft/styxlib/l5/messages/base/StyxRSingleQIDMessage.java
package com.v2soft.styxlib.l5.messages.base;

import com.v2soft.styxlib.l5.structs.StyxQID;

public class StyxRSingleQIDMessage extends StyxMessage {
    protected final StyxQID mQID;

    public StyxRSingleQIDMessage(int type, int tag, StyxQID qid) {
        super(type, tag);
        mQID = qid;
    }

    /**
     * @return QID structure
     */
    public StyxQID getQID() {
        return mQID;
    }

    @Override
    public String toString() {
        return String.format("%s\tQID: %s", super.toString(), getQID().toString());
    }
}
